vaults/azurekeyvault.go

```diff
@@ -81,8 +81,18 @@ func NewAKVSecretConfig(vaultURL, msiClientID, vaultAADResourceURL string) (*AKV
 		secretVersion = urlSegments[3]
 	}
 
+	vaultHostWithScheme := fmt.Sprintf("%s://%s", parsedURL.Scheme, parsedURL.Host)
+	splitStr := strings.SplitAfterN(vaultHostWithScheme, ".", 2)
+	// Ex. https://myacbvault.vault.azure.net -> ["https://myacbvault." "vault.azure.net"]
+	if len(splitStr) != 2 {
+		return nil, fmt.Errorf("extracted vault resource %s from resource ID %s is invalid", vaultHostWithScheme, vaultURL)
```

Comment:
nit: extracted from vault URL ?

Reply:
Thats because we pass vaultHostWithScheme to `VaultURL` below this line